packages/elements/src/combo-box/index.ts

```diff
@@ -279,13 +279,19 @@ export class ComboBox<T extends DataItem = ItemData> extends FormFieldElement {
     else {
       // Clone value arrays
       const newValues = values.slice(0, this.multiple ? values.length : 1);
-      const oldValues = this.composerValues.slice();
+      const oldValues = this.values.slice();
       // Create comparison strings to check for differences
       const newComparison = newValues.sort().toString();
       const oldComparison = oldValues.sort().toString();
-      // Should we update the selection state?
+
+      // Update the selection state when found new value
       if (newComparison !== oldComparison) {
         this.updateComposerValues(newValues);
+
+        if (this.freeText) {
+          this.freeTextValue = !this.composerValues.length ? newComparison : '';
```

Comment:
I think here is more correct to set newValues[0]: `this.freeTextValue = !this.composerValues.length ? newValues[0] : '';`

Reply:
Resolved